projects/plugins/jetpack/extensions/blocks/subscriptions/subscriptions.php

```diff
@@ -895,6 +897,30 @@ function maybe_gate_existing_comments( $comment ) {
 	return '';
 }
 
+/**
+ * Is the Jetpack_Token_Subscription_Service class loaded
+ *
+ * @return bool
+ */
+function is_jetpack_token_subscription_service_loaded(): bool {
+	return class_exists( 'Automattic\Jetpack\Extensions\Premium_Content\Subscription_Service\Jetpack_Token_Subscription_Service' );
+}
+
+/**
+ * Adds support for WP Super cache and Boost cache
+ */
+function maybe_prevent_super_cache_caching() {
+	// Prevents cached page to be served if the Membership cookie is present
+	if ( is_jetpack_token_subscription_service_loaded() ) {
+		do_action( 'wpsc_add_cookie', Jetpack_Token_Subscription_Service::JWT_AUTH_TOKEN_COOKIE_NAME );
+	}
+
+	if ( is_user_auth() ) {
+		// Do not cache the page if user is auth with Membership token
+		define( 'DONOTCACHEPAGE', true );
```

Comment:
Just on the off chance that another plugin has defined that constant already, it might be better to put an is_defined() check there too.